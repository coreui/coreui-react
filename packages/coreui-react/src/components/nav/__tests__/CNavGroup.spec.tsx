import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { CNavGroup } from '../index'

test('loads and displays CNavGroup component', async () => {
  const { container } = render(<CNavGroup toggler="anchorText" />)
  expect(container).toMatchSnapshot()
})

test('CNavGroup customize', async () => {
  const { container } = render(
    <CNavGroup className="bazinga" toggler="anchorText" visible={true} idx="1" />,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('nav-group')
  expect(container.firstChild).toHaveClass('bazinga')
  const arr = container.getElementsByClassName('nav-link')
  if (arr.length > 0) {
    //expect(arr[0].innerText).toHaveTextContent('anchorText')
    expect(arr[0].innerHTML).toBe('anchorText')
  } else {
    expect(true).toBe(false)
  }
})
