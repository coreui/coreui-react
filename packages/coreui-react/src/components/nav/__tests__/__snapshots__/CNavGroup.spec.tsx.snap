// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CNavGroup customize 1`] = `
<div>
  <li
    class="nav-group bazinga"
  >
    <a
      class="nav-link nav-group-toggle"
      href="#"
    >
      anchorText
    </a>
    <ul
      class="nav-group-items"
      style="height: 0px; display: block;"
    />
  </li>
</div>
`;

exports[`loads and displays CNavGroup component 1`] = `
<div>
  <li
    class="nav-group"
  >
    <a
      class="nav-link nav-group-toggle"
      href="#"
    >
      anchorText
    </a>
    <ul
      class="nav-group-items"
      style="height: 0px;"
    />
  </li>
</div>
`;
